apps/crawler/src/processors/block-processor.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Connection, Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+  EventHandlerContext,
+  SubstrateBlock,
+} from '@subsquid/substrate-processor';
+import { Block } from '@entities/Block';
+import { EventMethod, EventSection, ExtrinsicNames } from '@common/constants';
+import { ScanProcessor } from './scan-processor';
+import { SdkService } from '../sdk.service';
+import { ProcessorConfigService } from '../processor.config.service';
+
+const TRANSFER = `${EventSection.BALANCES}.${EventMethod.TRANSFER}`;
+const ENDOWED = `${EventSection.BALANCES}.${EventMethod.ENDOWED}`;
+
+@Injectable()
+export class BlockProcessor {
+  name = 'block';
+  private logger: Logger;
+  private processor: ScanProcessor;
+
+  constructor(
+    @InjectRepository(Block)
+    private modelRepository: Repository<Block>,
+    protected connection: Connection,
+    protected sdkService: SdkService,
+    private processorConfigService: ProcessorConfigService,
+  ) {
+    this.processor = new ScanProcessor(
+      this.name,
+      this.connection,
+      processorConfigService.getDataSource(),
+      processorConfigService.getRange(),
+      processorConfigService.getTypesBundle(),
+    );
+
+    this.logger = new Logger('BlockProcessor');
+
+    this.processor.addExtrinsicHandler(
+      ExtrinsicNames.TIMESTAMP_SET,
+      async (ctx) => {
+        await this.upsertHandler(ctx);
+      },
+    );
+
+    this.logger.log('Starting processor...');
+  }
+
+  public run(): void {
+    const params = this.processorConfigService.getAllParams();
+
+    this.logger.log({
+      msg: `Starting ${this.name} crawler...`,
+      params,
+    });
+
+    this.processor.run();
+  }
+
+  private async upsertHandler(ctx: EventHandlerContext): Promise<void> {
+    const { height, timestamp } = ctx.block;
+
+    const log = {
+      eventName: ExtrinsicNames.TIMESTAMP_SET,
+      blockNumber: height,
+      blockTimestamp: timestamp,
+    };
+
+    try {
+      const blockData = await this.getBlockData(ctx.block);
+
+      if (blockData) {
```

Comment:
What if no blockData? Now we just log that everything is ok. Maybe here we should log that something wrong? Or maybe remove this `if(blockData) {}`